File: 410-SplitArrayLargestSum/410-SplitArrayLargestSum.cpp
// Last updated: 28/06/2026, 01:03:16
class Solution {
public:
    bool numerofarraylimit(vector<int>& nums,int limit,int m)
    {
        int count=1;
        int n=nums.size();
        int sum=0;
        for(int i=0;i<n;i++)
        {
           if(nums[i]>limit)
            return false;
            if(nums[i]+sum>limit)
            {
                count++;
                sum=nums[i];
            }
            else
                sum+=nums[i];
        }
        return count<=m;
    }
    int splitArray(vector<int>& nums, int m) {
     int low=*max_element(nums.begin(),nums.end());  
        int high=0,mid;
        int n=nums.size();
        for(int i=0;i<n;i++)
            high+=nums[i];
        int ans=high;
        while(low<=high)
        {
           mid=low+(high-low)/2;
            if(numerofarraylimit(nums,mid,m)==false)
            {
                low=mid+1;
            }
            else
            {
                ans=mid;
                high=mid-1;
            }
        }
        return ans;
    }
};

File: 410-SplitArrayLargestSum/410-SplitArrayLargestSum.java
// Last updated: 28/06/2026, 01:03:14
class Solution {
    public int splitArray(int[] nums, int k) {
        int low=0,high=0;
        for(int i=0;i<nums.length;i++){
            low=Math.max(low,nums[i]);
            high+=nums[i];
        }
        int ans=high;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(isvalid(nums,mid,k)){
               ans=mid;
               high=mid-1;
            }
            else
                    low=mid+1;
        }
        return ans;
    }
    boolean isvalid(int[] nums,int mid,int k){
          int cnt=1,sum=0;
          for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            if(sum>mid){
                sum=nums[i];
                cnt++;
            }
          }
          if(cnt<=k)
               return true;
     return false;
    }
}